### PR TITLE
[master] Contract address eth style

### PR DESCRIPTION
**Pull request by n-hutton**
Previously there was a test for the contract addresses being the Eth style - this implements it proper.

Once the nonce fix is in I can test the combination of them both in Saeeds test framework or my test framework.

### CODE REVIEW

File: src/libServer/JSONConversion.cpp [hunk 1]
```diff
@@ -705,10 +705,10 @@ const Json::Value JSONConversion::convertTxtoEthJson(
   if (!txn.GetTransaction().GetCode().empty() &&
       IsNullAddress(txn.GetTransaction().GetToAddr())) {
     retJson["contractAddress"] =
-        "0x" +
-        Account::GetAddressForContract(txn.GetTransaction().GetSenderAddr(),
-                                       txn.GetTransaction().GetNonce() - 1)
-            .hex();
+        "0x" + Account::GetAddressForContract(
+                   txn.GetTransaction().GetSenderAddr(),
+                   txn.GetTransaction().GetNonce() - 1, TRANSACTION_VERSION_ETH)
```
Comment (valeryz) [inline src/libServer/JSONConversion.cpp:710]: Together with the Nonce Fix PR, should there be a  -1  here?

File: src/libServer/LookupServer.cpp [hunk 2]
```diff
@@ -801,7 +801,8 @@ std::pair<std::string, unsigned int> LookupServer::CheckContractTxnShards(
   Address affectedAddress =
       (Transaction::GetTransactionType(tx) == Transaction::CONTRACT_CREATION)
           ? Account::GetAddressForContract(tx.GetSenderAddr(),
-                                           tx.GetNonce() - 1)
+                                           tx.GetNonce() - 1,
```
Comment (valeryz) [inline src/libServer/LookupServer.cpp:804]: same

File: src/libServer/LookupServer.cpp [hunk 3]
```diff
@@ -909,7 +910,9 @@ Json::Value LookupServer::CreateTransaction(
                                    toAccountExist, toAccountIsContract);
         ret["Info"] = check.first;
         ret["ContractAddress"] =
-            Account::GetAddressForContract(fromAddr, tx.GetNonce() - 1).hex();
+            Account::GetAddressForContract(fromAddr, tx.GetNonce() - 1,
```
Comment (valeryz) [inline src/libServer/LookupServer.cpp:913]: same

File: src/libServer/LookupServer.cpp [hunk 4]
```diff
@@ -2127,7 +2131,8 @@ string LookupServer::GetContractAddressFromTransactionID(const string& tranID) {
                              "ID is not a contract txn");
     }
 
-    return Account::GetAddressForContract(tx.GetSenderAddr(), tx.GetNonce() - 1)
+    return Account::GetAddressForContract(tx.GetSenderAddr(), tx.GetNonce() - 1,
+                                          tx.GetVersionIdentifier())
```
Comment (valeryz) [inline src/libServer/LookupServer.cpp:2135]: same

**Review by valeryz — APPROVED** [review]

File: src/libEth/Eth.cpp [hunk 5]
```diff
@@ -59,7 +59,12 @@ Json::Value populateReceiptHelper(std::string const &txnhash, bool success,
   ret["root"] =
       "0x0000000000000000000000000000000000000000000000000000000000001010";
   ret["status"] = success ? "0x1" : "0x0";
-  ret["to"] = to;
+  if (to.empty()) {
```
Comment (n-hutton) [inline src/libEth/Eth.cpp:62]: this has to be null according to the eth spec.

**Pull request merged**